include/fmt/format.h

```diff
@@ -1570,91 +1570,107 @@ inline auto find_escape(const char* begin, const char* end)
  */
 #define FMT_STRING(s) FMT_STRING_IMPL(s, fmt::compile_string, )
 
-template <typename Char, typename OutputIt>
-auto write_escaped_string(OutputIt out, basic_string_view<Char> str)
-    -> OutputIt {
-  return copy_str<Char>(str.data(), str.data() + str.size(), out);
-}
-
 template <typename OutputIt, typename Char>
 auto write_escaped_cp(OutputIt out, const find_escape_result<Char>& escape)
     -> OutputIt {
   auto c = static_cast<Char>(escape.cp);
   switch (escape.cp) {
   case '\n':
-    *out++ = '\\';
-    c = 'n';
+    *out++ = static_cast<Char>('\\');
+    c = static_cast<Char>('n');
     break;
   case '\r':
-    *out++ = '\\';
-    c = 'r';
+    *out++ = static_cast<Char>('\\');
+    c = static_cast<Char>('r');
     break;
   case '\t':
-    *out++ = '\\';
-    c = 't';
+    *out++ = static_cast<Char>('\\');
+    c = static_cast<Char>('t');
     break;
   case '"':
     FMT_FALLTHROUGH;
   case '\'':
     FMT_FALLTHROUGH;
   case '\\':
-    *out++ = '\\';
+    *out++ = static_cast<Char>('\\');
     break;
   default:
     if (is_utf8()) {
       if (escape.cp < 0x100) {
-        return format_to(out, FMT_STRING("\\x{:02x}"), escape.cp);
+        *out++ = static_cast<Char>('\\');
+        *out++ = static_cast<Char>('x');
+        constexpr size_t len = 2;
+        Char buf[len];
+        fill_n(buf, len, static_cast<Char>('0'));
+        format_uint<4>(buf, escape.cp, len);
+        out = copy_str<Char>(buf, buf + len, out);
+        return out;
```

Comment:
Let's move this logic into a separate function and reuse here and three places below (prefix x/u and width can be parameters of this function).